perf/src/sigverify.rs

```diff
@@ -141,6 +142,62 @@ fn verify_packet(packet: &mut Packet) {
         pubkey_start = pubkey_end;
         sig_start = sig_end;
     }
+
+    check_for_simple_vote_transaction(packet, msg_start).ok();
+}
+
+fn check_for_simple_vote_transaction(
+    packet: &mut Packet,
+    msg_start_offset: usize,
+) -> Result<(), PacketError> {
+    let message_account_keys_len_offset = msg_start_offset
+        .checked_add(MESSAGE_HEADER_LENGTH)
+        .ok_or(PacketError::InvalidLen)?;
+
+    let (pubkey_len, pubkey_len_size) =
+        decode_shortu16_len(&packet.data[message_account_keys_len_offset..])
+            .map_err(|_| PacketError::InvalidLen)?;
+
+    let pubkey_start = message_account_keys_len_offset
+        .checked_add(pubkey_len_size)
+        .ok_or(PacketError::InvalidLen)?;
+
+    let instructions_len_offset = pubkey_len
+        .checked_mul(size_of::<Pubkey>())
+        .and_then(|v| v.checked_add(pubkey_start))
+        .and_then(|v| v.checked_add(size_of::<Hash>()))
+        .ok_or(PacketError::InvalidLen)?;
+
+    let (instruction_len, instruction_len_size) =
+        decode_shortu16_len(&packet.data[instructions_len_offset..])
+            .map_err(|_| PacketError::InvalidLen)?;
+
+    // skip if has more than 1 instruction
+    if instruction_len != 1 {
+        return Ok(());
+    }
+
+    let instruction_start = instructions_len_offset
+        .checked_add(instruction_len_size)
+        .ok_or(PacketError::InvalidLen)?;
+
+    let (instruction_id_index, _size) = decode_shortu16_len(&packet.data[instruction_start..])
```

Comment:
1. Need to ensure that `instruction_start < packet.meta.size` before this access
2. Instead of `instruction_id_index`, can you name it `instruction_program_id_index`?

Reply:
Good call, changed.